src/storage/entry/option.rs

```diff
@@ -0,0 +1,193 @@
+use core::marker::PhantomData;
+
+use crate::storage::entry;
+use crate::{key::Key, storage::OptionStorage};
+use option_bucket::{NoneBucket, OptionBucket, SomeBucket};
+
+struct VacantEntryNone<'this, K: Key, V> {
+    _key: PhantomData<K>,
```

Comment:
This `PhantomData` is not needed. It's only used to implement the `fn key()` fn which can instead be inlined in `impl VacantEntry` instead.